File: Essilfie/Day 12/product_of_array_except_self.rb
# @param {Integer[]} nums
# @return {Integer[]}

def product_except_self(nums)
    res = [1] * nums.size

    prefix = 1
    for i in 0..nums.length-1
        res[i] *= prefix
        prefix *= nums[i]
    end
    
    postfix = 1
    for i in (0..nums.length-1).to_a.reverse
        res[i] *= postfix
        postfix *= nums[i]
    end
    
    return res
end

File: Zeuhz/Day 13 - Product of Array Except Self/productOfArrayExceptSelf.js
const productExceptSelf = (nums) => {
  let product = [],
    count = 1;

  for (let i = 0; i < nums.length; i++) {
    product[i] = count;
    count *= nums[i];
  }

  count = 1;

  for (let i = nums.length - 1; i >= 0; i--) {
    product[i] *= count;
    count *= nums[i];
  }

  return product;
};
